_posts/2020-08-05-#19160.md

```diff
@@ -12,20 +12,20 @@ commit: 0602f87
+These review club notes are for PR [#19160 multiprocess: Add basic spawn and IPC support](https://github.com/bitcoin/bitcoin/pull/19160) ([branch](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-echo.7)), focused on the second eponymous commit: [`aa4d626`](https://github.com/bitcoin-core-review-club/bitcoin/commit/aa4d626). _IPC_ in the title is short for [Inter-process communication](https://en.wikipedia.org/wiki/Inter-process_communication).
 
 The key things this commit adds are an [`IpcProcess::spawn()`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L24-L26) method, which a bitcoin parent process can call to spawn a new child process, and an [`IpcProcess::serve()`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L31-L33) method, which a bitcoin child process can call [after](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/bitcoind.cpp#L180-L185) [startup](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/bitcoin-wallet.cpp#L82-L88) to communicate back to its parent process.
 
-In follow-up PR [#10102 Multiprocess bitcoin](https://github.com/bitcoin/bitcoin/pull/10102), this functionality is used so a `bitcoin-gui` process can spawn a `bitcoin-node` process, and a `bitcoin-node` process can spawn a `bitcoin-wallet` process, and gui, node, and wallet functionality can run in separate processes that are protected from each other. In further follow-ups [#19460 Add bitcoin-wallet -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19460) and [#19461 Add bitcoin-gui -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19461), more flexibility is added so node, wallet, and GUI processes can be started and stopped independently.
+In follow-up PR [#10102 Multiprocess bitcoin](https://github.com/bitcoin/bitcoin/pull/10102), this functionality is used so a `bitcoin-gui` process can spawn a `bitcoin-node` process, and a `bitcoin-node` process can spawn a `bitcoin-wallet` process, and GUI, node, and wallet functionality can run in separate processes that are protected from each other. In further follow-ups [#19460 Add bitcoin-wallet -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19460) and [#19461 Add bitcoin-gui -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19461), more flexibility is added so node, wallet, and GUI processes can be started and stopped independently.
 
-All communication between Bitcoin Core processes happens though [internal C++ interfaces](https://github.com/bitcoin/bitcoin/tree/master/src/interfaces), which are just C++ classes with pure virtual methods. The virtual methods allow bitcoin GUI, node, and wallet code to be written in a straightforward way that doesn't require dealing with complications of IPC. IPC is handled by the multiprocess framework added in this PR, to avoid the need to complicate application code with low level I/O.
```

Comment:
I think this can stay as a link to bitcoin/bitcoin (better would be to pin it to a specific commit hash)

Reply:
Linked to https://github.com/bitcoin/bitcoin/blob/f4cfa6d/doc/files.md and fixed up though->through

Reply:
Wait, wrong link

Reply:
Linked to https://github.com/bitcoin/bitcoin/tree/f4cfa6d/src/interfaces